server/src/main/java/org/opensearch/index/store/remote/utils/TransferManager.java

```diff
@@ -95,6 +95,12 @@ public IndexInput fetchBlob(BlobFetchRequest blobFetchRequest) throws IOExceptio
     @SuppressWarnings("removal")
     private static FileCachedIndexInput createIndexInput(FileCache fileCache, StreamReader streamReader, BlobFetchRequest request) {
         try {
+            // This local file cache is ref counted and may not strictly enforce configured capacity.
+            // If we find available capacity is exceeded, deny further BlobFetchRequests.
+            if (fileCache.capacity() < fileCache.usage().usage() && !fileCache.canOverflow()) {
+                fileCache.prune();
+                throw new IOException("Local file cache capacity exceeded - BlobFetchRequest failed: " + request.getFilePath());
```

Comment:
Maybe include the capacity and usage numbers in the exception message?

```suggestion
                throw new IOException("Local file cache capacity (" + fileCache.capacity() + ") exceeded (" + fileCache.usage().usage() + ") - BlobFetchRequest failed: " + request.getFilePath());
```